h/api/models/test/elastic_test.py

```diff
@@ -7,117 +10,463 @@
+
+        expected = [DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'https://example.com',
+                                 'type': 'rel-canonical',
+                                 'created': None, 'updated': None}),
+
+                    DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'http://example.com',
+                                 'type': 'self-claim',
+                                 'created': None, 'updated': None})]
+
+        assert doc.uris.sort() == expected.sort()
+
+    @pytest.mark.parametrize('doc', [
+        Document({'highwire': {'doi': 'foobar'}}, claimant='http://example.com'),
+        Document({'highwire': {'doi': 'doi:foobar'}}, claimant='http://example.com')])
+    def test_uris_generates_doi_uri_from_highwire_meta(self, doc):
+        expected = [DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'doi:foobar',
+                                 'type': 'highwire-doi',
+                                 'created': None, 'updated': None}),
+
+                    DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'http://example.com',
+                                 'type': 'self-claim',
+                                 'created': None, 'updated': None})]
+
+        assert doc.uris.sort() == expected.sort()
+
+    @pytest.mark.parametrize('doc', [
+        Document({'dc': {'doi': 'foobar'}}, claimant='http://example.com'),
+        Document({'dc': {'doi': 'doi:foobar'}}, claimant='http://example.com')])
+    def test_uris_generates_doi_uri_from_dc_meta(self, doc):
+        expected = [DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'doi:foobar',
+                                 'type': 'dc-doi',
+                                 'created': None, 'updated': None}),
+
+                    DocumentURI({'claimant': 'http://example.com',
+                                 'uri': 'http://example.com',
+                                 'type': 'self-claim',
+                                 'created': None, 'updated': None})]
+
+        assert doc.uris.sort() == expected.sort()
+
+
+class TestDocumentMeta(object):
+    def test_created(self):
+        meta = DocumentMeta({'created': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert meta.created == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_updated(self):
+        meta = DocumentMeta({'updated': datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)})
+        assert meta.updated == datetime.datetime(2016, 2, 25, 16, 45, 23, 371848)
+
+    def test_claimant(self):
+        meta = DocumentMeta({'claimant': 'http://example.com'})
+        assert meta.claimant == 'http://example.com'
+
+    def test_claimant_normalized(self):
+        meta = DocumentMeta({'claimant': 'http://example.com'})
```

Comment:
Want to check that it actually gets normalised? (Don't bother mocking out the normaliser... just stick a trailing slash on it or something.)